src/chromo/models/pythia8.py

```diff
@@ -161,28 +173,61 @@ def _cross_section(self, kin=None):
             diffractive_axb=st.sigmaAXB,
         )
 
+        # TODO use pythia.info.sigmaGen(i) for cross-sections including nucleus in the
+        # initial collision system.
+        #   pythia.info.sigmaTot() are the estimated cross-section values for the
+        # equivalent pp collision for each sub-processes.
+        #   pythia.info.sigmaGen(i) are the cross-section values generated by the MC
+        # events for each subprocesses. Best is to grab these values at the end of the
+        # event generation when the full statistics is available.
+        # The subprocesses are defined with codes in Pythia:
+        #    i   subprocess
+
+    #   101  non-diffractive
+    #   102  A B -> A B elastic
+    #   103  A B -> X B single diffractive
+    #   104  A B -> A X single diffractive
+    #   105  A B -> X X double diffractive
+    #   106  A B -> A X B central diffractive
+
     def _set_kinematics(self, kin):
         config = self._config[:]
 
+        # Path prefix for some cache files that Pythia8 generates
+        # to speed up subsequent runs
+        cache_prefix = str(
+            _cache_base_dir()
+            / "Pythia8"
+            / f"cache_{int(kin.p1)}_{int(kin.p2)}_{kin.ecm / MeV:.0f}".replace("-", "m")
+        )
+
         # TODO use numpy PRNG instead of Pythia's
         config += [
             # use center-of-mass frame
             "Beams:frameType = 1",
             # do not print progress
             "Next:numberCount = 0",
+            "MultipartonInteractions:reuseInit = 3",
+            f"MultipartonInteractions:initFile = {cache_prefix}.mpi",
```

Comment:
Is this globally necessary or recommended for all pythia8 users or just nuclei? If this is only relevant for users of nuclei then it needs to be further below.

Reply:
The initialization of the MPI framework is the most time-consuming part of the whole initialization step, when simulating collisions for heavy ion systems with Angantyr. It is a feature for the users to gain time if they run the same (projectile, target, eCM) combination for more than one time.

Reply:
If you want to look up the details of this MPI initialization, here is the documentation: https://pythia.org/latest-manual/MultipartonInteractions.html (scroll down to "Initialization speed-up").

Reply:
Short summary: yes, it is useful for everyone and therefore this placement is ok.

Reply:
OK, so I read through the docs and I disagree with that it is safe to use this globally. Changes to any of the parameters except a few would collide with that file. Also there are clear limitations when using variable beam energy etc. such as the `Beams:allowIDAswitch`. Catching all of the potential issues and misuse is *not* the task of `chromo` and has to be redirected to the user, and the Pythia docs.

Therefore, I favor that Pythia8 is initialized with its default values. The version with accelerated initialization should be either explained in an example `example_speedup_pythia8_initialization.ipynb` or in a separate subclass `Pythia8Fast` or so.